src/System.Net.Http/tests/StressTests/HttpStress/ClientOperations.cs

```diff
@@ -149,21 +179,55 @@ public static class ClientOperations
                     using (var req = new HttpRequestMessage(HttpMethod.Get, "/headers") { Version = httpVersion })
                     using (HttpResponseMessage m = await ctx.SendAsync(req))
                     {
-                        ValidateResponse(m, httpVersion);
+                        ValidateHttpVersion(m, httpVersion);
+                        ValidateStatusCode(m);
                         ValidateContent(ctx.ContentSource, await m.Content.ReadAsStringAsync());
                     }
                 }),
 
+                ("GET Echo Headers",
+                async ctx =>
+                {
+                    Version httpVersion = ctx.GetRandomHttpVersion();
+
+                    using (var req = new HttpRequestMessage(HttpMethod.Get, "/echoHeaders") { Version = httpVersion })
+                    {
+                        ctx.PopulateWithRandomHeaders(req.Headers);
+
+                        using (HttpResponseMessage res = await ctx.SendAsync(req))
+                        {
+                            ValidateHttpVersion(res, httpVersion);
+                            ValidateStatusCode(res);
+
+                            // Validate that response headers are being echoed
```

Comment:
Did you mean that "request" headers are being echoed?

Reply:
Indeed I did. Good catch